### PR TITLE
Test Django 5.0

### PR DESCRIPTION
**Pull request by adamchainz**
## Problem

Test the new version of Django that’s in beta.

Depends on #582. If you want to consider updating `SaveSignalHandlingModel`, we can look at that, but I really suggest that it be removed.

## Solution

Added classifier to `setup.py`, new test environment to `tox.ini`, setup, and upgraded test requirements to fix some warnings.

## Commandments

- [x] Write PEP8 compliant code.
- [x] Cover it with tests.
- [x] Update `CHANGES.rst` file to describe the changes, and quote according issue with `GH-<issue_number>`.
- [x] Pay attention to backward compatibility, or if it breaks it, explain why.
- [x] Update documentation (if relevant).

### CODE REVIEW

**Comment by adamchainz** [comment]
Apologies for the initially wrong title, that will have caused duplicate-looking threads on email.

**Review by Mogost — APPROVED** [review]

**Comment by shanx** [comment]
Hey @adamchainz could you possible merge `master` into your branch and update the django 5.0 version to the release one. Then I would say let's go ahead with merging this PR.

File: tox.ini
Comment (shanx) [inline tox.ini:29]: Would pin this to minimum django 5.0 release version

Comment (adamchainz) [reply tox.ini]: Sure, now it’s out :)

**Comment by adamchainz** [comment]
I rebased rather than merging!

**Comment by shanx** [comment]
Hey @adamchainz thanks for the work. I just merged your PR and tried to push a new release to pypi. It seems there is a release gh workflow: https://github.com/jazzband/django-model-utils/actions/runs/7856043884/job/21438400809 but it doesn't seem to work. Do you happen to know more?

**Comment by adamchainz** [comment]
I have no idea, sorry. I'm no maintainer.

**Comment by shanx** [comment]
hey @carljm, are you the original author of this library? After merging Adam's work I tried to push out a new release, but it looks like the release script doesn't fully work anymore:  https://github.com/jazzband/django-model-utils/actions/runs/7856043884/job/21438400809  do you have any idea wat could go wrong here?

**Comment by carljm** [comment]
@shanx I am the original author, but I have neither used the project or been involved in maintenance for at least 8 years. I'm afraid I don't know anything about how that publish script works or what went wrong with it. I am willing to collaborate as needed in handoff steps to a responsible maintainer, but beyond that I don't have any interest in maintaining django-model-utils. Sorry!

**Comment by shanx** [comment]
I totally understand that @carljm. I thought the spend a bit of time on this, get a release out there. But also don't have time to really dive into things. I see jazzband has the possibility to get help on a new release to pypi, I'll give that a shot. Anyway thanks for answering and have a beautiful day.

**Comment by shanx** [comment]
@adamchainz a release was just pushed to pypi by @charettes. Thanks everyone!

**Pull request merged**
